### PR TITLE
Add move plugin

### LINKED ISSUE #533: Feature: Move plugin
Add new plugin which will be move specified fields to the target field in a certain mode (`allow|block`).
> In `allow` mode, the specified `fields` will be moved;
> in `block` mode, the unspecified `fields` will be moved.

Config:
```
Fields []cfg.FieldSelector
Mode string
Target cfg.FieldSelector
```

### PR DESCRIPTION
**Pull request by kirillov6**
# Description

Added new move plugin.

Fixes #533

### CODE REVIEW

File: plugin/action/move/move.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+func factory() (pipeline.AnyPlugin, pipeline.AnyConfig) {
+	return &Plugin{}, &Config{}
+}
+
+func (p *Plugin) Start(config pipeline.AnyConfig, _ *pipeline.ActionPluginParams) {
+	p.config = config.(*Config)
+
+	p.invalidConfig = !p.config.isValid()
+	if p.invalidConfig {
+		return
```
Comment (vadimalekseev) [inline plugin/action/move/move.go:165]: Let's use fatal here

File: plugin/action/move/move.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+			targetNode = targetNode.AddFieldNoAlloc(event.Root, tField).MutateToObject()
+			continue
+		}
+
+		node := targetNode.Dig(tField)
+		if node == nil {
+			node = targetNode.AddFieldNoAlloc(event.Root, tField).MutateToObject()
+			continueDig = false
+		} else if !node.IsObject() {
+			return pipeline.ActionPass
```
Comment (vadimalekseev) [inline plugin/action/move/move.go:198]: Let's overwrite the existing field

File: plugin/action/move/move.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+		}
+	} else {
+		for _, node := range event.Root.AsFields() {
+			value := node.AsFieldValue()
+			if value == targetNode {
+				continue
+			}
+
+			name := node.AsString()
+			if slices.IndexFunc(p.fields, func(f []string) bool { return f[0] == name }) == -1 {
```
Comment (vadimalekseev) [inline plugin/action/move/move.go:222]: I think we should use map here

**Review by vadimalekseev — APPROVED** [review]
LGTM!

File: plugin/action/move/move.go (hunk 1 above)
Comment (kirillov6) [reply plugin/action/move/move.go]: done

File: plugin/action/move/move.go (hunk 2 above)
Comment (kirillov6) [reply plugin/action/move/move.go]: done

File: plugin/action/move/move.go (hunk 3 above)
Comment (kirillov6) [reply plugin/action/move/move.go]: done

File: plugin/action/move/move.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.ActionPluginParams) {
+	p.config = config.(*Config)
+	if err := p.config.validate(); err != nil {
+		params.Logger.Fatalf("invalid config: %s", err.Error())
+	}
+
+	p.fields = make(map[string][]string)
+	isBlockMode := p.config.Mode == modeBlock
+	for _, fs := range p.config.Fields {
+		// in `block` mode, max field depth is 1
```
Comment (vano144) [inline plugin/action/move/move.go:170]: let's add it to doc

**Review by vano144 — APPROVED** [review]

File: plugin/action/move/move.go (hunk 4 above)
Comment (kirillov6) [reply plugin/action/move/move.go:170]: it's already in doc, similar for `target` field
image

**Pull request merged**